com.ibm.streamsx.topology/opt/python/include/splpy_setup.h

```diff
@@ -131,11 +140,23 @@ class SplpySetup {
           SplpyGeneral::loadFunction("streamsx.spl.types", "_get_timestamp_tuple"));
        SplpyGeneral::decimalClass(
           SplpyGeneral::loadFunction("decimal", "Decimal"));
+   }
+
+   static void setupSplNull() {
+       SplpyGIL lock;
 
-       PyObject *fn = SplpyGeneral::loadFunction("streamsx.spl.types", "null");
-       SplpyGeneral::isSplNull(
-          SplpyGeneral::pyCallObject(fn, NULL));
-       Py_DECREF(fn);
+       // Get a pointer to SPL null.
+       PyObject *null = SplpyGeneral::callFunction("streamsx.spl.types", "null", NULL, NULL);
+
+        // Call isSplNull() passing in a pointer to SPL null which will
+        // be the first caller (as this is in setup)
+        // and thus set the local pointer to SPL null.
+        bool in = SplpyGeneral::isSplNull(null);
+        if (!in) {
+          throw SplpyGeneral::generalException("setup",
+                        "Internal error - SPL null handling");
+        }
+        Py_DECREF(null);
```

Comment:
Need to not decrement the reference. The function `isSplNull` holds onto the reference so it needs to have a count associated with it.